google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddRemarketingAction.java

```diff
@@ -0,0 +1,152 @@
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+
+    // Creates a remarketing action with the specified name.
+    RemarketingAction remarketingAction =
+        RemarketingAction.newBuilder()
+            .setName(StringValue.of("Remarketing action #" + System.currentTimeMillis()))
+            .build();
+
+    // Creates a remarketing action operation.
+    RemarketingActionOperation operation =
+        RemarketingActionOperation.newBuilder().setCreate(remarketingAction).build();
+
+    // Issues a mutate request to add the remarketing action and prints out some information.
+    String remarketingActionResourceName;
+    try (RemarketingActionServiceClient conversionActionServiceClient =
+        googleAdsClient.getLatestVersion().createRemarketingActionServiceClient()) {
+      MutateRemarketingActionsResponse response =
+          conversionActionServiceClient.mutateRemarketingActions(
+              Long.toString(customerId), Collections.singletonList(operation));
+      remarketingActionResourceName = response.getResults(0).getResourceName();
+      System.out.printf(
+          "Added remarketing action with resource name '%s'.%n", remarketingActionResourceName);
+    }
+
+    // Creates a query that retrieves the previously created remarketing action with its generated
+    // tag snippets.
+    String query =
+        String.format(
+            "SELECT remarketing_action.id, remarketing_action.name,"
+                + " remarketing_action.tag_snippets "
+                + "FROM remarketing_action "
+                + "WHERE remarketing_action.resource_name = '%s'",
+            remarketingActionResourceName);
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues a search request.
+      SearchPagedResponse searchPagedResponse =
+          googleAdsServiceClient.search(Long.toString(customerId), query);
+
+      // There is only one row because we limited the search using the resource name, which is
+      // unique.
+      GoogleAdsRow googleAdsRow = searchPagedResponse.iterateAll().iterator().next();
+
+      // Prints some attributes of the remarketing action. The ID and tag snippets are generated by
+      // the API.
+      RemarketingAction newRemarketingAction = googleAdsRow.getRemarketingAction();
+      System.out.printf(
+          "Remarketing action has ID %d and name '%s'.%n%n",
+          newRemarketingAction.getId().getValue(), newRemarketingAction.getName().getValue());
```

Comment:
I think it would be more readable with one argument per line as done in other code examples.

Reply:
This is the format produced by the Google Java formatter, so would prefer to leave this as-is. Otherwise, any manual edits I make will probably be undone by the next editor of this file.

Reply:
SG